### ISSUE TITLE
Issue with lib setting in the tsconfig.json

### ISSUE THREAD

**Issue by remojansen** [TypeStrong/tsify#196]
So I'm targeting ES5 but I have added support for some ES6 elements using `lib` in `tsconfig`:

```js
{
    "compilerOptions": {
        "target": "es5",
        "lib": ["es6", "dom"],
        ...
    }
}
```

You can see my [tsconfig.json](https://github.com/remojansen/InversifyJS/blob/master/tsconfig.json#L4) if you need it.

I'm using `typescript@rc` 2.0 and this is my gulp task:

```js
gulp.task("build-bundle-test", function() {

  var mainTsFilePath = "test/inversify.test.ts";
  var outputFolder   = "temp/";
  var outputFileName = "bundle.test.js";

  var bundler = browserify({
    debug: true,
    standalone : "inversify"
  });

  // TS compiler options are in tsconfig.json file
  return bundler.add(mainTsFilePath)
                .plugin(tsify, { typescript: require("typescript") })
                .bundle()
                .pipe(source(outputFileName))
                .pipe(buffer())
                .pipe(sourcemaps.init({ loadMaps: true }))
                .pipe(sourcemaps.write("."))
                .pipe(gulp.dest(outputFolder));
});
```

You can see my [gulpfile.js](https://github.com/remojansen/InversifyJS/blob/master/gulpfile.js#L157-L165) if you need it.

I run the build in AppVeyor and is failing only in one version of node:

```
nodejs_version=stable
```

You can see all the build logs [here](https://ci.appveyor.com/project/remojansen/inversifyjs/build/186) and the failing build [here](https://ci.appveyor.com/project/remojansen/inversifyjs/build/186/job/seme7r35dlq5rs87#L812).

The error is:
```
TypeScript error: Error TS2318: Cannot find global type 'Array'.
```

This maybe think that it could be related with the `lib` setting in the `tsconfig.json` file but I'm not sure.

Thanks!

**Comment by cartant** [comment]
@basarat Are you able to suggest what might be causing this particular TypeScript error? The version of Node.js that seems to be the problem is 6.5.0. As the OP has stated, the builds are passing on several other Node.js versions.

**Comment by remojansen** [comment]
It looks like is behaving [like if I used `--noLib` option](https://github.com/Microsoft/TypeScript/issues/3930) but is weird because it only does that in one version of node.

**Comment by remojansen** [comment]
I've been testing and it looks like tsify uses `noResolve` even when I set it to false?

**Comment by cartant** [comment]
The options passed to `createProgram` definitely do not contain `noResolve`. However, when running your bundling with the debug logging enabled:

    NODE_DEBUG=tsify,tsify-trace gulp build-bundle-test

there seems to be something amiss with the lib paths:

    TSIFY-TRACE 10776: Parsing c:/lib.es2015.d.ts
    TSIFY-TRACE 10776: Parsing c:/lib.dom.d.ts

Which could have a similar effect to `noResolve`, as the files would essentially be left out of the compilation. I'm looking into it.

**Comment by remojansen** [comment]
Thanks a lot for your help :)

**Comment by cartant** [comment]
I've found the problem. There is a new method in the compile host - `getDefaultLibLocation` - and if it's not there, TypeScript attempts to determine the location using the host's `getDefaultLibFileName` method. However, they way it does that doesn't work with the configuration that `tsify` uses.

This will be patched pretty soon.

Strangely, I cannot see how this could possibly be working on other versions of Node.js. I tried it on a VM with 4.5.0 and it failed. So your passing builds are something of a mystery.

**Comment by remojansen** [comment]
I tried in my laptop (OSX with node v5.8.0) and it was fine. Tried then with node v6.5.0 and it is fine too. There must be something weird in AppVeyor.

**Comment by cartant** [comment]
My laptop and VM are both Windows boxes. It looks like Windows is the common denominator.

I cannot say that I'm surprised; dealing with backslash replacement is an ongoing source of frustration with the TypeScript compiler. It looks like there's yet another place in which TypeScript's `normalizeSlashes` function needs to be called.

However, I still cannot see why it wouldn't fail on the AppVeyor builds that used other versions of Node.js.

**Comment by cartant** [comment]
Yep, calling `normalizeSlashes` on the result of `getDefaultLibFileName` also fixes the problem.

**Comment by cartant** [comment]
This should be resolved by #197. I've published 1.0.6 to NPM.

**Comment by remojansen** [comment]
Thanks a lot! 😄
